core/embed/rust/src/ui/model_mercury/component/instructions.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{
+    strutil::TString,
+    ui::{
+        component::{text::TextStyle, Component, Event, EventCtx, Never},
+        constant::WIDTH,
+        geometry::{Alignment, Offset, Rect},
+        model_mercury::theme,
+        shape::{Renderer, Text},
+    },
+};
+
+/// Component showing a task instruction (e.g. "Swipe up") and optionally task
+/// description (e.g. "Confirm transaction") to a user. The component
+/// is typically placed at the bottom of the screen. The height of the provided
+/// area must be 18px (only instruction) or 37px (both description and
+/// instruction). The content and style of both description and instruction is
+/// configurable separatedly.
+pub struct Instructions<'a> {
+    area: Rect,
+    text_instruction: TString<'a>,
+    text_description: Option<TString<'a>>,
+    style_instruction: &'static TextStyle,
+    style_description: &'static TextStyle,
+}
+
+impl<'a> Instructions<'a> {
+    /// height for component with only instruction [px]
+    pub const HEIGHT_SIMPLE: i16 = 18;
+    /// height for component with both description and instruction [px]
+    pub const HEIGHT_DEFAULT: i16 = 37;
+
+    pub fn new<T: Into<TString<'a>>>(instruction: T) -> Self {
+        Self {
+            area: Rect::zero(),
+            text_instruction: instruction.into(),
+            text_description: None,
+            style_instruction: &theme::TEXT_SUB,
+            style_description: &theme::TEXT_SUB,
+        }
+    }
+
+    pub fn with_description<T: Into<TString<'a>>>(self, description: T) -> Self {
+        Self {
+            text_description: Some(description.into()),
+            ..self
+        }
+    }
+
+    pub fn update_instruction<T: Into<TString<'a>>>(&mut self, s: T) {
+        self.text_instruction = s.into();
+    }
+
+    pub fn update_description<T: Into<TString<'a>>>(&mut self, s: T) {
+        self.text_description = Some(s.into());
+    }
+
+    pub fn update_instruction_style(&mut self, style: &'static TextStyle) {
+        self.style_instruction = style;
+    }
+
+    pub fn update_description_style(&mut self, style: &'static TextStyle) {
+        self.style_description = style;
+    }
+}
+
+impl<'a> Component for Instructions<'a> {
+    type Msg = Never;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        let h = bounds.height();
+        assert!(h == Instructions::HEIGHT_SIMPLE || h == Instructions::HEIGHT_DEFAULT);
+        assert!(bounds.width() == WIDTH);
```

Comment:
Appreciate the defensiveness here but in the past it has paid off to do the right thing for other sizes when it's not much work - e.g. model T firmware works fairly well on [STM32F429I-DISC1](https://docs.trezor.io/trezor-firmware/core/misc/disc1.html) despite its non-square screen. My suggestion is to remove the asserts and either fill the entire bounds or split off a fixed height rectangle from them.

Reply:
I actually don't see a problem here with using this component on different devices.
The `assert`s ensure that a parent component provided rectangle of certain height (18px or 37px) spanned over the whole width (`WIDTH` it taken from `model_xyz/constant.rs`). So the only problem on bigger screen is that the instruction would be unnecessarily small at the bottom.